src/dsp_tools/dsp_tools.py

```diff
@@ -198,9 +198,9 @@ def program(user_args: list[str]) -> None:
                               verbose=args.verbose)
     elif args.action == "xmlupload":
         if args.validate:
-            success = validate_xml_against_schema(input_file=args.xmlfile)
+            success = validate_xml_against_schema(xml_file_as_path_or_parsed=args.xmlfile)
         else:
-            success = xml_upload(input_file=args.xmlfile,
+            success = xml_upload(xml_file_as_path_or_parsed=args.xmlfile,
```

Comment:
Maybe I'm lacking context here, as I'm not involved in the mass upload project. But to me, "parsed" is not self explanatory, so the renaming of those variables doesn't add any clarity for me, but makes it a lot harder to read.  
And in terms of the code context here: this is a function of the highest abstraction level, called from the program's entry point. For that reason I don't think that the parameter name here should have to reflect the format of the input. So IMO this renaming of the parameter is not an improvement.

Reply:
Thanks for pointing this out. I reverted this change